src/main/resources/avro/sequenceAnnotations.avdl

```diff
@@ -0,0 +1,137 @@
+@namespace("org.ga4gh.models")
+/**
+This protocol defines annotations on GA4GH genomic sequences It includes two
+types of annotations: continuous and discrete hierarchical.
+
+The discrete hierarchical annotations are derived from the Sequence Ontology
+(SO) and GFF3 work 
+
+   http://www.sequenceontology.org/gff3.shtml
+
+The goal is to be able to store annotations using the GFF3 and SO conceptual
+model, although there is not necessarly a one-to-one mapping in Avro records
+to GFF3 records.
+
+The minimum requirement is to be able to accurately represent the current
+state of the art annotation data and the full SO model.  Feature is the
+core generic record which corresponds to the a GFF3 record.
+*/
+protocol SequenceAnnotations {
+
+  import idl "common.avdl";
+  import idl "metadata.avdl";
+
+  /**
+  Type defining a collection of attributes associated with various protocol
+  records.  Each attribute is a name that maps to an array of one or more
+  values.  Values can be strings, external identifiers, or ontology terms.
+  Values should be split into the array elements instead of using a separator
+  syntax that needs to parsed.
+  */
+  // TODO: how are multiple instances of a given attribute vs multiple values
+  // for an attribute distinguished
+  record Attributes {
+    map<array<union {string, ExternalIdentifier, OntologyTerm}>> vals = {};
+  }
+  
+  /**
+  Node in the annotation graph that annotates a contiguous region of a
+  sequence.
+  */
+  record Feature {
+    /**
+    Id of this annotation node.
+    */
+    string id;
+
+    /**
+    Parent Id of this node. Set to empty string if node has no parent.
+    */
+    string parentId;
+
+    /**
+    Ordered array of Child Ids of this node.
+    Since not all child nodes are ordered by genomic coordinates,
+    this can't always be reconstructed from parentId's of the children alone.
+    */
+
+    array<string> childIds = [];
+
+    /**
+    Identifier for the containing feature set.
+    */
+    string featureSetId;
+
+    /**
+    The reference on which this feature occurs.
+    (e.g. `chr20` or `X`)
+    */
+    string referenceName;
```

Comment:
@macieksmuga: I agree that we should plan for sequence graphs. In fact, this is exactly why I want hashes exclusively: it is the only identifier that we can be assured will 1) exist and 2) be immutable. Ultimately, all features *must* be on a linear reference ultimately otherwise the notion of a start and end are meaningless. 

A hash represents a path that is composed of sequence segments, which may also be hashed. It just so happens that the smallest genomic segment that we understand right now is a chromosome. In the future, we could easily imagine features being on small segments that we refer to by hashes. In other words, I think that human-assigned accession or name impedes graph adoption, and using a hash enables graph adoption.

But, you guys have the sequence graph brains... you should decide. (But, I'm skeptical that the right answer is sequence name or accession.)